tests/metrics/functional/test_classification.py

```diff
@@ -0,0 +1,296 @@
+                     torch.tensor([int(bool(idx % 2)) for idx in range(100)]),
+                     torch.ones(100), 1., 40
+                     ),
+        pytest.param(torch.randint(low=51, high=99, size=(100,), dtype=torch.float) / 100,
+                     torch.tensor([int(bool(idx % 2)) for idx in range(100)]),
+                     None, 1., 39
+                     )
+    ]
+)
+@pytest.mark.usefixtures("random")
+def test_binary_clf_curve(pred, target, sample_weight, pos_label, exp_shape):
+    fps, tps, thresh = _binary_clf_curve(pred, target, sample_weight, pos_label)
+    assert isinstance(tps, torch.Tensor)
+    assert isinstance(fps, torch.Tensor)
+    assert isinstance(thresh, torch.Tensor)
+    assert tps.shape == (exp_shape,)
+    assert fps.shape == (exp_shape,)
+    assert thresh.shape == (exp_shape,)
+
+
+@pytest.mark.parametrize(['pred', 'target', 'expected_p', 'expected_r', 'expected_t'], [
+    pytest.param(torch.tensor([1, 2, 3, 4]), torch.tensor([1, 0, 0, 1]),
+                 torch.tensor([0.5, 1 / 3, 0.5, 1., 1.]), torch.tensor([1, 0.5, 0.5, 0.5, 0.]),
+                 torch.tensor([1, 2, 3, 4]))
+])
+def test_pr_curve(pred, target, expected_p, expected_r, expected_t):
+    p, r, t = precision_recall_curve(pred, target)
+    assert p.size() == r.size()
+    assert p.size(0) == t.size(0) + 1
+
+    assert torch.allclose(p, expected_p.to(p))
+    assert torch.allclose(r, expected_r.to(r))
+    assert torch.allclose(t, expected_t.to(t))
+
+
+@pytest.mark.parametrize(['pred', 'target', 'expected_tpr', 'expected_fpr'], [
+    pytest.param(torch.tensor([0, 1]), torch.tensor([0, 1]), torch.tensor([0, 1, 1]),
+                 torch.tensor([0, 0, 1])),
+    pytest.param(torch.tensor([1, 0]), torch.tensor([0, 1]), torch.tensor([0, 0, 1]),
+                 torch.tensor([0, 1, 1])),
+    pytest.param(torch.tensor([1, 1]), torch.tensor([1, 0]), torch.tensor([0, 1]),
+                 torch.tensor([0, 1])),
+    pytest.param(torch.tensor([1, 0]), torch.tensor([1, 0]), torch.tensor([0, 1, 1]),
+                 torch.tensor([0, 0, 1])),
+    pytest.param(torch.tensor([0.5, 0.5]), torch.tensor([0, 1]), torch.tensor([0, 1]),
+                 torch.tensor([0, 1]))
+])
+def test_roc_curve(pred, target, expected_tpr, expected_fpr):
+    fpr, tpr, thresh = roc(pred, target)
+
+    assert fpr.shape == tpr.shape
+    assert fpr.size(0) == thresh.size(0)
+    assert torch.allclose(fpr, expected_fpr.to(fpr))
+    assert torch.allclose(tpr, expected_tpr.to(tpr))
+
+
+@pytest.mark.parametrize(['pred', 'target', 'expected'], [
+    pytest.param(torch.tensor([0, 1]), torch.tensor([0, 1]), 1.),
+    pytest.param(torch.tensor([1, 0]), torch.tensor([0, 1]), 0.),
+    pytest.param(torch.tensor([1, 1]), torch.tensor([1, 0]), 0.5),
+    pytest.param(torch.tensor([1, 0]), torch.tensor([1, 0]), 1.),
+    pytest.param(torch.tensor([0.5, 0.5]), torch.tensor([1, 0]), 0.5)
+])
+def test_auroc(pred, target, expected):
```

Comment:
can we maybe also add a test for auc separately instead of only testing it indirectly @cuent ?